include/boost/geometry/algorithms/detail/buffer/buffered_piece_collection.hpp

```diff
@@ -293,7 +276,7 @@ struct buffered_piece_collection
 
     // Specificly for offsetted rings around points
     // but also for large joins with many points
-    typedef geometry::sections<normal_box_type, 2> sections_type;
+    typedef geometry::sections<box_type, 2> sections_type;
```

Comment:
@barendgehrels I stumbled upon a problem regarding this change. The sections and their boxes are used in partition:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/algorithms/detail/buffer/buffered_piece_collection.hpp#L592
and then deeper here:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/algorithms/detail/buffer/get_piece_turns.hpp#L224-L225
and then here:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/algorithms/detail/buffer/get_piece_turns.hpp#L161-L164
then here:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/algorithms/detail/sections/section_functions.hpp#L136
which as you can see from the parameters relies on the fact that both points and boxes are robust.
And finally here:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/algorithms/detail/sections/section_functions.hpp#L136
which causes robust point be compared with non-robust boxes.

Now, I didn't check how sections/boxes are created. If boxes are created from robust points and then simply the value is stored using `double` then I guess we can loose precision when integer values are compared with doubles in which case the bounding box may not cover the robust point (maybe?). If they are created from original points then comparing them with robust points doesn't make any sense because robust coordinates are rescaled.

From the values I see:
```
point            | {6.3920000000000003, 2.5440000000000000}    | boost::geometry::model::point<double,2,boost::geometry::cs::cartesian>
robust_point     | {152743, -763011}                           | boost::geometry::model::point<__int64,2,boost::geometry::cs::cartesian>
other_robust_box | {{-428079.00000000012, -554791.00000000012},
                    {428086.00000000012, -554790.99999999988}} | boost::geometry::model::box<boost::geometry::model::point<double,2,boost::geometry::cs::cartesian> >
point_robust_box | {{-360956.00000000006, -763011.00000000012},
                    {152743.00000000003, -78078.999999999985}} | boost::geometry::model::box<boost::geometry::model::point<double,2,boost::geometry::cs::cartesian> >
```
I think it is the former, so normal point is passed here, sections use the same coordinate type as the normal point (double) but they hold the rescaled values. If that's the case then changing `sections_type` to use `box_type` instead of `robust_box_type` was wrong in the worst case and useless in the best case because values are still rescaled anyway.

This is non-detectable for native integer robust types but if you wanted to use MultiPrecision integral type as robust type then the less-comparison between MultiPrecision integral and `double` does not exist. I have a branch where this is done if you want to play with it.

What do you think we should do here?

All of this can be reproduced e.g. with `buffer_multi_polygon.cpp` test case.

P.S. I also don't understand why does buffer deal with sections manually like that. Partition should do the space partitioning and call visitor accordingly. An algorithm should not interfere with that. But that's a different story.

Reply:
Hi Adam, thanks, I think I recently saw this too and fixed it in my next PR, which is not yet ready.
I’ll come back to it next Wednesday.

Reply:
I can reproduce it, but it's a different thing than I thought. I'm working on it.